src/DurableTask.SqlServer/Scripts/logic.sql

```diff
@@ -1,6 +1,85 @@
 ﻿-- Copyright (c) Microsoft Corporation.
 -- Licensed under the MIT License.
 
+-- Create custom types. This must be done before creating stored procedures.
+-- IMPORTANT: If you make any changes to these types, you must also add a line in the schema upgrade script
+--            (for example, schema-1.X.0) to drop them so that those changes are properly reflected.
+IF TYPE_ID(N'__SchemaNamePlaceholder__.InstanceIDs') IS NULL
+    CREATE TYPE __SchemaNamePlaceholder__.InstanceIDs AS TABLE (
+        [InstanceID] varchar(100) NOT NULL
+    )
+GO
+
+IF TYPE_ID(N'__SchemaNamePlaceholder__.MessageIDs') IS NULL
+    -- WARNING: Reordering fields is a breaking change!
+    CREATE TYPE __SchemaNamePlaceholder__.MessageIDs AS TABLE (
+        [InstanceID] varchar(100) NULL,
+        [SequenceNumber] bigint NULL
+    )
+GO
+
+IF TYPE_ID(N'__SchemaNamePlaceholder__.HistoryEvents') IS NULL
+    -- WARNING: Reordering fields is a breaking change!
+    CREATE TYPE __SchemaNamePlaceholder__.HistoryEvents AS TABLE (
+        [InstanceID] varchar(100) NULL,
+        [ExecutionID] varchar(50) NULL,
+        [SequenceNumber] bigint NULL,
+        [EventType] varchar(40) NULL,
+        [Name] varchar(300) NULL,
+        [RuntimeStatus] varchar(30) NULL,
+        [TaskID] int NULL,
+        [Timestamp] datetime2 NULL,
+        [IsPlayed] bit NULL,
+        [VisibleTime] datetime2 NULL,
+        [Reason] varchar(max) NULL,
+        [PayloadText] varchar(max) NULL,
+        [PayloadID] uniqueidentifier NULL,
+        [ParentInstanceID] varchar(100) NULL,
+        [Version] varchar(100) NULL,
+        [TraceContext] varchar(800) NULL
```

Comment:
Each of these type definitions previously lived in schema-1.0.0.sql. I realized, however, that they needed to be moved to logic.sql so that we can more easily update them when changes are needed (like adding `TraceContext` properties). The only changes are to HistoryEvents, OrchestrationEvents, and TaskEvents, where we add the `TraceContext` property to the end of the list of properties.